cranelift-codegen/meta/src/shared/instructions.rs

```diff
@@ -993,6 +993,32 @@ pub fn define(
         .operands_out(vec![addr]),
     );
 
+    // Note this instruction is marked as having other side-effects, so GVN won't try to hoist it,
+    // which would result in it being subject to spilling. It's free to do so, because this
```

Comment:
Good call. Tweaked it to "While not hoisting would generally hurt performance, since a computed value used many times may need to be regenerated before each use, it is not the case here: this instruction doesn't generate any code. That's because, by definition the pinned register is entirely user controlled, so there's nothing to do to reset it."